File: sorting_algo/1-algo.c
#include <stdio.h>

void selectionSort(int arr[], int n)
{
    int i, j, minIndex, temp;

    for (i = 0; i < n - 1; i++)
    {
        minIndex = i;

        for (j = i + 1; j < n; j++)
        {
            if (arr[j] < arr[minIndex])
                minIndex = j;
        }

        temp = arr[minIndex];
        arr[minIndex] = arr[i];
        arr[i] = temp;
    }
}

void printarray(int arr[], int size)
{
    int i;

    for (i = 0; i < size; i++)
        printf("%d ", arr[i]);
    printf("\n");
}

int main(void)
{
    int arr[] = {12, 11, 13, 5, 6};
    int n = sizeof(arr) / sizeof(arr[0]);

    printf("Original array: \n");
    printarray(arr, n);

    selectionSort(arr, n);

    printf("Sorted array: \n");
    printarray(arr, n);

    return (0);
}

File: sorting_algo/1-algo.py
#!/usr/bin/python3


def selectionSort(n, lst=None):
    if lst is None:
        lst = []

    for i in range(n):
        minIndex = i

        for j in range(i + 1, n):
            if lst[j] < lst[minIndex]:
                minIndex = j

        temp = lst[minIndex]
        lst[minIndex] = lst[i]
        lst[i] = temp

def printlst(size, lst=[]):
    for i in range(size):
        print(f'{lst[i]} ', end='')
    print()

if __name__ == '__main__':
    lst = [12, 11, 13, 5, 6]
    n = len(lst)

    print('Original array:')
    printlst(n, lst)

    selectionSort(n, lst)

    print('Sorted array:')
    printlst(n, lst)
